apps/functions-subscription/src/adapters/azure/functions/list-trials.ts

```diff
@@ -0,0 +1,72 @@
+import * as H from '@pagopa/handler-kit';
+import { pipe, flow } from 'fp-ts/lib/function';
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import * as E from 'fp-ts/Either';
+import * as O from 'fp-ts/lib/Option';
+import * as t from 'io-ts';
+import { httpAzureFunction } from '@pagopa/handler-kit-azure-func';
+import { TrialPaginatedCollection } from '../../../generated/definitions/internal/TrialPaginatedCollection';
+import { SystemEnv } from '../../../system-env';
+import { getAndValidateUser, parseQueryParameter } from './middleware';
+import { toHttpProblemJson } from './errors';
+import { toTrialListAPI } from './codec';
+import { TrialIdCodec } from '../../../domain/trial';
+import {
+  IntegerFromString,
+  WithinRangeInteger,
+} from '@pagopa/ts-commons/lib/numbers';
+
+type Env = Pick<SystemEnv, 'listTrials'>;
+
+export const QueryPageSize = t.union([
+  t.undefined,
+  // cast the value to an integer and ensure it's within the 1 to 100 range (1 <= value < 101)
+  IntegerFromString.pipe(WithinRangeInteger(1, 101)),
+]);
+
+const makeHandlerKitHandler: H.Handler<
+  H.HttpRequest,
+  | H.HttpResponse<TrialPaginatedCollection>
+  | H.HttpResponse<H.ProblemJson, H.HttpErrorStatusCode>,
+  Env
+> = H.of((req: H.HttpRequest) =>
+  pipe(
+    RTE.ask<Env>(),
+    RTE.apFirst(RTE.fromEither(getAndValidateUser(['ApiTrialSupport'])(req))),
+    RTE.apSW(
+      'pageSize',
+      pipe(
+        parseQueryParameter(QueryPageSize, 'pageSize')(req),
+        // set a default pageSize if the value received from the request is null or undefined
```

Comment:
```suggestion
        // set a default pageSize if the value received from the request is null or undefined (default value specified in OpenAPI specification)
```

Reply:
Updated in cb9047422e69c939846d97666d4205a5bd3635c8